freud/locality.pyx

```diff
@@ -459,16 +458,16 @@ cdef class NeighborList:
                 Array of per-bond weights (if :code:`None` is given, use a
                 value of 1 for each weight) (Default value = :code:`None`).
         """
-        index_i = freud.common.convert_array(index_i, 1, dtype=np.uint64)
-        index_j = freud.common.convert_array(index_j, 1, dtype=np.uint64)
+        index_i = freud.common.convert_array(index_i, (None,), dtype=np.uint64)
+        index_j = freud.common.convert_array(index_j, (None,), dtype=np.uint64)
```

Comment:
We could get rid of the shape check below by enforcing `index_j` has the same shape as `index_i`.